src/Compilers/VisualBasic/Portable/Symbols/Tuples/TupleErrorFieldSymbol.vb

```diff
@@ -94,7 +94,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
 
         Public Overloads Function Equals(other As TupleErrorFieldSymbol) As Boolean
             Return other Is Me OrElse
-                (other IsNot Nothing AndAlso Me._tupleFieldId = other._tupleFieldId AndAlso Me.ContainingType = other.ContainingType)
+                (other IsNot Nothing AndAlso
+                Me._tupleElementIndex = other._tupleElementIndex AndAlso
+                Me.ContainingType = other.ContainingType AndAlso
+                Me.Name = other.Name)
```

Comment:
>Me.Name = other.Name [](start = 16, length = 20)

This might call a VB runtime helper, consider explicitly calling comparison API on string with explicit comparison options.